Documentation/install/guides/kube-router.rst

```diff
@@ -0,0 +1,116 @@
+.. _kops_guide:
+
+****************************
+Using kube-router to run BGP
+****************************
+
+This guide explains how to configure Cilium and kube-router to co-operate to
+use kube-router for BGP peering and route propagation and Cilium for policy
+enforcement and load-balancing.
+
+.. note::
+
+    This feature is regarded as ``tech-preview``. Please provide feedback and
+    file a GitHub issue if you experience any problems.
+
+Deploy kube-router
+##################
+
+Download the kube-router DaemonSet template:
+
+.. code:: bash
+
+    curl -LO https://raw.githubusercontent.com/cloudnativelabs/kube-router/master/daemonset/generic-kuberouter-only-advertise-routes.yaml
+
+Open the file ``generic-kuberouter-only-advertise-routes.yaml`` and edit the
+``args:`` section. The following arguments are **requried** to be set to
+exactly these values:
+
+.. code:: bash
+
+    - "--run-router=true"
+    - "--run-firewall=false"
+    - "--run-service-proxy=false"
+    - "--enable-cni=false"
+    - "--enable-pod-egress=false"
+
+The following arguments are **optional** and may be set according to your
+needs. Please see the `kube-router user guide
+<https://github.com/cloudnativelabs/kube-router/blob/master/docs/user-guide.md>`_
+for more information:
+
+.. code:: bash
+
+    - "--enable-ibgp=false"
+    - "--enable-overlay=false"
+    - "--peer-router-ips=<CHANGE ME>"
+    - "--peer-router-asns=<CHANGE ME>"
+    - "--cluster-asn=<CHANGE ME>"
+
+Apply the DaemonSet file to deploy kube-router and verify it has come up
+correctly:
+
+.. code:: bash
+
+    $ kubectl apply -f generic-kuberouter-only-advertise-routes.yaml
+    $ kubectl -n kube-system get pods -l k8s-app=kube-router
+    kube-router-2dgkt   1/1       Running   0          52s
+    kube-router-pl9j6   1/1       Running   0          52s
+    kube-router-shgk8   1/1       Running   0          52s
+    kube-router-wvpm5   1/1       Running   0          52s
+
+Deploy Cilium
+#############
+
+In order for route installation to be delegated to kube-router,
+tunneling/encapsulation must be disabled. This is done by changing the
+``tunnel`` key in the ConfigMap ``cilium-config`` or by adjusting the
+DaemonSet to set ``--tunnel=disabled``:
+
+.. code:: bash
+
+    # Encapsulation mode for communication between nodes
+    # Possible values:
+    #   - disabled
+    #   - vxlan (default)
+    #   - geneve
+    tunnel: "disabled"
+
+.. note::
+
+    If you are running kube-route with overlay enabled, adjust for the
+    encapsulation overhead by setting the environment variable ``MTU`` to the
+    value 1450.
```

Comment:
Where? How? Please indicate this to the users

Reply:
This needs to be reworked anyway to use a new option. Will add clearer documentation.